### PR TITLE
Add forall and exists to Array and EmbedArray

### PR DESCRIPTION
**Pull request by arosenberger**
Implements #80 

Review by @densh

### CODE REVIEW

File: tests/src/test/scala/EmbedArraySuite.scala [hunk 1]
```diff
@@ -127,4 +127,40 @@ class EmbedArraySuite extends FunSuite {
     assert(EmbedArray.empty[EPoint].filter(_ => true).isEmpty)
     assert(EmbedArray.empty[EPoint].filter(_ => false).isEmpty)
   }
+
+  test("forall") {
+    val arr = Array(
```
Comment (densh) [inline tests/src/test/scala/EmbedArraySuite.scala]: `EmbedArray` here?

Comment (densh) [inline tests/src/test/scala/EmbedArraySuite.scala]: And everywhere else in this test suite.

Comment (arosenberger) [inline tests/src/test/scala/EmbedArraySuite.scala]: Ah good catch. Will fix these up.

**Comment by arosenberger** [comment]
@densh All fixed up

**Comment by densh** [comment]
LGTM

**Pull request merged**